Plugins/Sycl/src/Seeding/CreateSeedsForGroupSycl.cpp

```diff
@@ -113,77 +99,128 @@ void createSeedsForGroupSycl(
+    if (!device_resource){
+      bottomSPsView = vecmem::get_data(bottomSPs);
+      topSPsView = vecmem::get_data(topSPs);
+      middleSPsView = vecmem::get_data(middleSPs);
+    } else {
+      deviceBottomSPs = std::make_unique<vecmem::data::vector_buffer
+                                          <detail::DeviceSpacePoint>>
+                                              (B, *device_resource);
+      deviceTopSPs = std::make_unique<vecmem::data::vector_buffer
+                                          <detail::DeviceSpacePoint>>
+                                              (T, *device_resource);
+      deviceMiddleSPs = std::make_unique<vecmem::data::vector_buffer
+                                          <detail::DeviceSpacePoint>>
+                                              (M, *device_resource);
+
+      copy(vecmem::get_data(bottomSPs), *deviceBottomSPs);
+      copy(vecmem::get_data(topSPs), *deviceTopSPs);
+      copy(vecmem::get_data(middleSPs), *deviceMiddleSPs);
+
+      bottomSPsView = vecmem::get_data(*deviceBottomSPs);
+      topSPsView = vecmem::get_data(*deviceTopSPs);
+      middleSPsView = vecmem::get_data(*deviceMiddleSPs);
+    }
     //*********************************************//
     // ********** DUPLET SEARCH - BEGIN ********** //
     //*********************************************//
 
+    // Create the output data of the duplet search - jagged vectors.
+    std::unique_ptr<vecmem::data::jagged_vector_buffer<uint32_t>> midBotDupletBuffer;
+    std::unique_ptr<vecmem::data::jagged_vector_buffer<uint32_t>> midTopDupletBuffer;
+    if (!device_resource) {
+      midBotDupletBuffer = std::make_unique<vecmem::data::jagged_vector_buffer<uint32_t>>
+                                                (std::vector<std::size_t>(M, 0),
+                                                  std::vector<std::size_t>(M,B),
+                                                  resource);
+      midTopDupletBuffer = std::make_unique<vecmem::data::jagged_vector_buffer<uint32_t>>
+                                                (std::vector<std::size_t>(M, 0),
+                                                  std::vector<std::size_t>(M, T),
+                                                  resource);
+    } else {
+      midBotDupletBuffer = std::make_unique<vecmem::data::jagged_vector_buffer<uint32_t>>
+                                                (std::vector<std::size_t>(M, 0),
+                                                  std::vector<std::size_t>(M,B),
+                                                  *device_resource, &resource);
+      midTopDupletBuffer = std::make_unique<vecmem::data::jagged_vector_buffer<uint32_t>>
+                                                (std::vector<std::size_t>(M, 0),
+                                                  std::vector<std::size_t>(M, T),
+                                                  *device_resource, &resource);
+    }
```

Comment:
Since we are setting up the buffer here anyway, the following formalism could be a little more compact:

```c++
     midBotDupletBuffer = std::make_unique<vecmem::data::jagged_vector_buffer<uint32_t>>
                                                (std::vector<std::size_t>(M, 0),
                                                  std::vector<std::size_t>(M,B),
                                                  (device_resource ? *device_resource : resource),
                                                  (device_resource ? &resource : nullptr));
```

But the current code is correct as well. :wink:

Reply:
Done.